src/env.rs

```diff
@@ -83,35 +83,26 @@ impl Source for Environment {
         let mut m = HashMap::new();
         let uri: String = "the environment".into();
 
-        let separator = match self.separator {
-            Some(ref separator) => separator,
-            _ => "",
-        };
-
-        let group_separator = match self.separator {
-            Some(ref separator) => separator,
-            _ => "_",
-        };
+        let separator = self.separator.as_deref().unwrap_or("");
+        let group_separator = self.separator.as_deref().unwrap_or("_");
```

Comment:
This is nice, but it would require that we (and our users) would use rust 1.40.0 or newer.
So I would like to have a CI job for that version before I merge this piece.

I will see if I can get to it myself in the next days, just leaving this as a comment and self-reminder!